triton/client/decode_manifest_triton.py

```diff
@@ -0,0 +1,416 @@
+    --num-tasks 300 \
+    --manifest-filename /mnt/samsung-t7/yuekai/aishell-test-dev-manifests/data/fbank/aishell_cuts_test.jsonl.gz
+
+# For streaming wenet server
+./decode_manifest_triton.py \
+    --server-addr localhost \ 
+    --streaming \
+    --compute-cer \
+    --context 7 \
+    --model-name streaming_wenet \
+    --num-tasks 300 \
+    --manifest-filename /mnt/samsung-t7/yuekai/aishell-test-dev-manifests/data/fbank/aishell_cuts_test.jsonl.gz
+
+
+
+(Note: You have to first start the server before starting the client)
+"""
+
+import argparse
+import sys
+from functools import partial
+import asyncio
+import time
+from pathlib import Path
+import types
+import math
+
+import numpy as np
+from icefall.utils import store_transcripts, write_error_stats
+from lhotse import CutSet, load_manifest
+
+import tritonclient
+import tritonclient.grpc.aio as grpcclient
+from tritonclient.utils import np_to_triton_dtype, InferenceServerException
+
+DEFAULT_MANIFEST_FILENAME = "/mnt/samsung-t7/yuekai/aishell-test-dev-manifests/data/fbank/aishell_cuts_test.jsonl.gz"  # noqa
+
+def get_args():
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+
+    parser.add_argument(
+        "--server-addr",
+        type=str,
+        default="localhost",
+        help="Address of the server",
+    )
+
+    parser.add_argument(
+        "--server-port",
+        type=int,
+        default=8001,
+        help="Port of the server",
+    )
+
+    parser.add_argument(
+        "--manifest-filename",
+        type=str,
+        default=DEFAULT_MANIFEST_FILENAME,
+        help="Path to the manifest for decoding",
+    )
+
+    parser.add_argument(
+        "--model-name",
+        type=str,
+        default="transducer",
+        help="triton model_repo module name to request",
+    )
+
+    parser.add_argument(
+        "--num-tasks",
+        type=int,
+        default=50,
+        help="Number of tasks to use for sending",
+    )
+
+    parser.add_argument(
+        "--log-interval",
+        type=int,
+        default=5,
+        help="Controls how frequently we print the log.",
+    )
+
+    parser.add_argument(
+        "--compute-cer",
+        action="store_true",
+        default=False,
+        help="""True to compute CER, e.g., for Chinese.
+        False to compute WER, e.g., for English words.
+        """,
+    )
+
+    parser.add_argument(
+        "--streaming",
+        action="store_true",
+        default=False,
+        help="""True for streaming ASR.
+        """,
+    )
+
+    parser.add_argument('--chunk_size',
+                        type=int,
+                        required=False,
+                        default=16,
+                        help='chunk size default is 16')
+
+    parser.add_argument('--context',
+                        type=int,
+                        required=False,
+                        default=97,
+                        help='subsampling context')
```

Comment:
What does this mean? Is the value after subsampling or before subsampling?

Reply:
Sorry, it's a bug. It didn't cause any issue since we send first chunk audios which are longer than the server needed.

After this fix, it sends exactly the length of audio the server needs.

It may be possible to change it using padding methods when we init the feature_extractor part.